src/main/groovy/uk/gov/hmrc/jenkinsjobbuilders/domain/builder/JobBuilder.groovy

```diff
@@ -166,6 +177,10 @@ final class JobBuilder implements Builder<Job> {
                     environmentVariablesScriptContent, environmentVariablesGroovyScript))
         }
 
+        if (!this.preScmSteps.isEmpty()) {
```

Comment:
Groovy truth will ensure that the condition only evaluates to true is the list is not null and is not empty, so you could replace with this:

```suggestion
        if (this.preScmSteps) {
```